### DISCUSSION TITLE
Support searching multiple collections based on a name pattern (glob-like)

### DISCUSSION THREAD

**Discussion by igoristic** [qdrant/qdrant#2013] [Ideas]
Elasticsearch has a feature where we can [search multiple indices](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/multi-index.html) (like collections in Qdrant) with a single query. Example:

If we have the following collections:
```
20230101_cats
20230201_cats
20230101_dogs
20230201_dogs
```

And we do a search on pattern like this:
```
202301*,-*_dogs
```
it will only search in one collection:
```
20230101_cats
```
Since the pattern states to include everything that matches `202301*`, and exclude (notice the `-`) everything that matches `*_dogs`

---

Another example:

```
*_dogs
```

Would only search in the following collections:
```
20230101_dogs
20230201_dogs
```

---

I wrote something similar for Kibana a while back that might give you a good idea about the logic: https://github.com/igoristic/kibana/blob/1e5e9876666ee14cd869e4f80706e348218066d8/x-pack/plugins/monitoring/common/es_glob_patterns.ts

Thank you Qdrant team, for such an awesome product and constant releases 🙇 ❤️  ❤️

**Comment by kinoute** [comment]
Any updates on this? We could make an issue as "feature request" to talk about this. It would be a great addition!

**Comment by timvisee** [comment]
In Qdrant, Multitenancy is done in a single collection through payloads.

This, and reasoning behind it, is described here: https://qdrant.tech/documentation/guides/multiple-partitions/
